### DISCUSSION TITLE
No Speed Up by MKL (Matrix Diagonalisation)

### DISCUSSION THREAD

**Discussion by dvdllrch** [QuSpin/QuSpin#688] [Q&A]
I want to compute the ground state of huge system and tried taking advantage of OMP and MKL. While OMP is cutting my computational time, MKL doesn't seem to have an effect. Any idea why?
It's an Intel CPU, Anaconda version is >= 2.5 (MKL exists in the environment, so I'm guessing Numpy is build against it) and I only tried with the example script, so it's not a problem of the code.

**Comment by mgbukov** [comment]
can you share you script?

OMP turns on for basis/operator construction and applications of matvec,
while MKL for default numoy/scipy operations such as diagonalization, etc

Marín

**Reply by dvdllrch** [reply]
It's just the example script under [Parallel Computing in QuSpin](https://quspin.github.io/QuSpin/examples/example12.html#example12-label). Eigenvalues take the same time to compute (even longer at times).

Thank you for the quick response!

**Reply by dvdllrch** [reply]
Could you give me an idea how much speed up I should expect for eigsh depending on the number of used cores?

**Comment by mgbukov** [comment]
> Could you give me an idea how much speed up I should expect for eigsh 
> depending on the number of used cores? 

I don't know what MKL speedup you should expect :(

but what I can say is that quspin.operators.hamiltonian.eigsh() is a 
trivial wrapper of scipy.sparse.linalg.eigsh so probably the info u need 
is mentioned somewhere on the web for scipy+MKL.
